src/wallet/walletServiceStorageProxy.ts

```diff
@@ -0,0 +1,186 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { IStorage, IHistoryTx, IAddressInfo, IAddressMetadata } from '../types';
+import Transaction from '../models/transaction';
+import HathorWalletServiceWallet from './wallet';
+import { FullNodeTxResponse } from './types';
+import transactionUtils from '../utils/transaction';
+
+/**
+ * Storage proxy that implements missing storage methods for wallet service
+ * by delegating to wallet service API calls.
+ *
+ * This proxy enables nano contract transaction signing by providing:
+ * - getAddressInfo: Maps addresses to BIP32 indices
+ * - getTx: Fetches transaction data from full node API
+ * - getTxSignatures: Delegates to transaction signing utilities
+ */
+export class WalletServiceStorageProxy {
+  private wallet: HathorWalletServiceWallet;
+
+  private originalStorage: IStorage;
+
+  constructor(wallet: HathorWalletServiceWallet, originalStorage: IStorage) {
+    this.wallet = wallet;
+    this.originalStorage = originalStorage;
+  }
+
+  /**
+   * Creates a proxy that wraps the original storage with additional methods
+   * needed for nano contract transaction signing.
+   */
+  createProxy(): IStorage {
+    return new Proxy(this.originalStorage, {
+      get: this.proxyHandler.bind(this),
+    });
+  }
+
+  private proxyHandler(target: IStorage, prop: string | symbol, receiver: IStorage): unknown {
```

Comment:
What's `symbol` and this `receiver`?

Reply:
Maybe a docstring could help explain those parameters.